content/docs/tech-insights/track-sonarqube/index.md

```diff
@@ -0,0 +1,135 @@
+2. Visualise the projects which are doing well, and also the worst offenders.
+3. Create a check that can tell us which software is not adhering to standards set by the organization.
+
+As we go through this process, you’ll learn:
+
+1. How to configure a component in Roadie with its Sonarqube project key
+2. How to configure and run the Roadie Sonarqube broker application to connect Sonarqube with Roadie
+3. How to create an HTTP Datasource to collect data from Sonarqube
+4. How to create a Check with Tech Insights
+
+Let’s get started.
+
+## Configure a Component with its Sonarqube project key
+
+1. First identify the project key for the catalog Component.
+    You can find this in the Sonarqube dashboard. Find the project, click Administration -> Update key and copy the "Project Key"
+2. Next edit the catalog YAML file for the Component 
+    The easiest way to do this is to find the component in the Roadie catalog and click the edit pencil in the top right hand side of the About Card.
+3. Edit and save the YAML file to include the `sonarqube.org/project-key` annotation.
+
+    ```yaml
+    ---
+    apiVersion: backstage.io/v1alpha1
+    kind: Component
+    metadata:
+      name: test-react-app
+      annotations:
+        sonarqube.org/project-key: punkle_test-react-app
+    ```
+
+## How to configure and run the Roadie Sonarqube broker application to connect Sonarqube with Roadie
+
+1. First generate a new token in the Sonarqube dashboard
+
+    Click on your profile in the top right of the application, click My Account. Then click Security. Enter a token name and click "Generate Token". Copy the token value.
+
+2. Run the Roadie broker client application to connect Sonarqube with backstage.
+
+    ```shell
+    docker run \
+      --env SONARQUBE_INSTANCE_URL=https://<sonarqube_address> \
+      --env BROKER_TOKEN=sonarqube \
+      --env BROKER_SERVER_URL=https://<tenant_name>.broker.roadie.so \
+      --env SONARQUBE_TOKEN=<sonarqube_token> \
+      roadiehq/broker:5691884253.43.1-sonarqube
+    ```
+
+3. Determine the IP address from where the broker client is running and set this IP in the broker allow list in Roadie.
+    
+    You can determine the IP address locally by running the following command. This command assumes you are running the docker container locally. When you production
+
+    ```bash
+    curl 'https://api.ipify.org'
+    ```
+   
+4. To set the IP allow list go to Administration > Settings > Broker enter the IP address and subnet mask using CIDR notation.
+
+5. Confirm that the broker client by looking at the docker container logs. You should see the following text:
+
+    ```text
+    successfully established a websocket connection to the broker server
+    ```
+
+## Create an HTTP Data Source to collect data from Sonarqube
+
+1. Within Roadie, click on Tech Insights -> Data Sources and click "Add Datasource"
+
+2. Set the name to "Sonarqube", and set the description "Sonarqube".
```

Comment:
If I worked in a customer company, I would like this description to be vastly more descriptive.

I would like something like "Collects data from our on-prem SonarQube instance".

Reply:
done.